src/vmm/src/devices/virtio/iovec.rs

```diff
@@ -846,8 +893,37 @@ mod verification {
         }
     }
 
+    fn create_iov_deque() -> IovDeque {
+        // SAFETY: safe because the layout has non-zero size
+        let mem = unsafe {
+            std::alloc::alloc(std::alloc::Layout::from_size_align_unchecked(
+                2 * PAGE_SIZE,
+                PAGE_SIZE,
+            ))
+        };
+        IovDeque {
+            iov: mem.cast(),
+            start: kani::any_where(|&start| start < FIRECRACKER_MAX_QUEUE_SIZE),
+            len: 0,
+        }
+    }
+
+    /// During `IovDeque::drop` we call `memunmap` to deallocate the backing memory of the
+    /// `IovDeque`. For the purpose of the proof we substitute the allocation with a normal
+    /// allocation, so deallocate the memory here.
+    pub fn drop_iov_deque(deque: &mut IovDeque) {
+        // We previously allocated this memory with a call to `std::alloc::alloc` with the same
+        // allocator
+        unsafe {
+            std::alloc::dealloc(
+                deque.iov.cast(),
+                std::alloc::Layout::from_size_align_unchecked(2 * PAGE_SIZE, PAGE_SIZE),
+            )
+        }
+    }
```

Comment:
You don't actually need to do this, kani has no concept of memory leaks

Reply:
I need to do something for the `munmap` failure, so I did this. Would you prefer that I just `std::mem::forget` `deque.iov`?

Reply:
Ah, yeah, just doing `std::mem::forget` is fine :)

Reply:
done